README.md

```diff
@@ -1,2 +1,24 @@
 # Slicer-SOFA
-3D Slicer extension to enable simulations using the SOFA framework. This software is distributed under a MIT License.
+
+![Slicer-SOFA Logo](./slicer-sofa-logo.png)
+
+## Introduction
+
+Slicer-SOFA is an open-source extension that seamlessly integrates the SOFA into 3D Slicer, a versatile platform for medical image analysis and visualization. This integration facilitates the development and execution of complex biomechanical models, enhancing applications in surgical planning, medical training, and biomedical research. By combining the strengths of both platforms, Slicer-SOFA aims to advance the field of medical simulation and provide a comprehensive tool for clinicians, researchers, and educators. This software is distributed under a MIT License.
+
+## Software Architecture
+
+![Slicer-SOFA Architecture](./slicer-sofa-architecture.png)
+
+The **Slicer-SOFA Extension** bridges the capabilities of the open-source 3D Slicer platform and the **SOFA Framework**, enabling advanced simulation-based applications. This integration leverages Slicer's Python environment and **SofaPython3**, providing a seamless interface between Slicer and SOFA. 
```

Comment:
Small suggstion: this could be a good place to have embedded links, like to SofaPython3 for easy access.